pages/workouts/[id].jsx

```diff
@@ -0,0 +1,54 @@
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import useSWR from "swr";
+
+export default function WorkoutDetailsPage() {
+  const router = useRouter();
+  const id = router.query.id;
+
+  const { data: workout, isLoading, error } = useSWR(`/api/workouts/${id}`);
+  if (isLoading) {
+    <main>loading</main>;
+  }
+  if (error) {
+    <main>error</main>;
+  }
+  if (!workout) {
+    return <main>loading workout</main>;
+  }
+  const { name, exercises } = workout;
+  const muscleGroupsInWorkout = new Set();
```

Comment:
```suggestion
  const muscleGroupsInWorkout = collectMuscleGroups(exercises);
```
Lets create a custom hook for this. So you wont have to always create new Sets :-)